### PR TITLE
Support StorageManager for S3 using minio SDK

### PR DESCRIPTION
**Pull request by shlomitk1**
Signed-off-by: Shlomit Koyfman <shlomitk@il.ibm.com>
Fixes https://github.com/fybrik/fybrik/issues/1852
- This PR uses Minio SDK to allocate/delete buckets and removes all references to datashim and Dataset custom resources.
- Changes to FybrikApplication status:
    - no need for DatasetRef since the resource does not exist
    - added an indication whether the storage should be freed upon deletion of FybrikApplication, or it is a persistent storage. 
    Previously, a label inside Dataset resource has been used for this purpose.
- No "late" checks of status before plotter creation. In case of allocation errors, no data path is built.

### CODE REVIEW

**Review by roytman — COMMENTED** [review]
I think this PR should update the docs/StorageManagement.md.
Especially "Currently, only S3 storage is supported. Both allocation and deletion of the storage (if temporary) is done using [Datashim](https://datashim.io/)."

**Comment by shlomitk1** [comment]
> I think this PR should update the docs/StorageManagement.md. Especially "Currently, only S3 storage is supported. Both allocation and deletion of the storage (if temporary) is done using [Datashim](https://datashim.io/)."

Removed the description of the current state.

**Comment by roytman** [comment]
> > I think this PR should update the docs/StorageManagement.md. Especially "Currently, only S3 storage is supported. Both allocation and deletion of the storage (if temporary) is done using [Datashim](https://datashim.io/)."
> 
> Removed the description of the current state.

We can add a reference to this PR at line 272

File: charts/fybrik-crd/templates/app.fybrik.io_fybrikapplications.yaml [hunk 1]
```diff
@@ -249,7 +249,7 @@ spec:
                     description: DatasetDetails holds details of the provisioned storage
                     properties:
                       datasetRef:
-                        description: Reference to a Dataset resource containing the request to provision storage
+                        description: deprecated
```
[the remark this answers is not in the document]
Comment (shlomitk1) [reply charts/fybrik-crd/templates/app.fybrik.io_fybrikapplications.yaml:252]: Done

**Comment by shlomitk1** [comment]
> 
> We can add a reference to this PR at line 272

Done

**Pull request merged**